tests/phpunit/tests/test-translated-post.php

```diff
@@ -99,6 +99,70 @@ public function test_delete_post_translation() {
 		$this->assertSame( 0, self::$model->post->get_translation( $de, 'fr' ) ); // fails
 	}
 
+	/**
+	 * Checks that the translations group are kept unique when linking several translations together.
+	 *
+	 * @ticket #2717 see {https://github.com/polylang/polylang-pro/issues/2717}.
+	 */
+	public function test_save_translations() {
+		$posts = self::factory()->post->create_translated(
+			array( 'lang' => 'en' ),
+			array( 'lang' => 'fr' ),
+			array( 'lang' => 'de' )
+		);
+
+		$terms = wp_get_object_terms( $posts, 'post_translations' );
+		$this->assertCount( 1, $terms );
+
+		$this->assertSame( $posts['en'], self::$model->post->get_translation( $posts['en'], 'en' ) );
+		$this->assertSame( $posts['fr'], self::$model->post->get_translation( $posts['fr'], 'fr' ) );
+		$this->assertSame( $posts['de'], self::$model->post->get_translation( $posts['de'], 'de' ) );
+
+		$this->assertSame( $posts['fr'], self::$model->post->get_translation( $posts['en'], 'fr' ) );
+		$this->assertSame( $posts['de'], self::$model->post->get_translation( $posts['en'], 'de' ) );
+
+		$this->assertSame( $posts['en'], self::$model->post->get_translation( $posts['fr'], 'en' ) );
+		$this->assertSame( $posts['de'], self::$model->post->get_translation( $posts['fr'], 'de' ) );
+
+		$this->assertSame( $posts['fr'], self::$model->post->get_translation( $posts['de'], 'fr' ) );
+		$this->assertSame( $posts['en'], self::$model->post->get_translation( $posts['de'], 'en' ) );
+
+		/*
+		.* Removes the translations from the group by updating the German post.
+		.* It keeps a translations group only for German.
+		.* Translations in English and French are also unlinked.
+		.* See `PLL_Translated_Post::delete_translation()`.
+		 */
+		self::$model->post->save_translations( $posts['de'], array() );
+
+		$terms = wp_get_object_terms( $posts, 'post_translations' );
+		$this->assertCount( 1, $terms );
+
+		$this->assertSame( 0, self::$model->post->get_translation( $posts['de'], 'fr' ) );
+		$this->assertSame( 0, self::$model->post->get_translation( $posts['de'], 'en' ) );
+
+		// Links again the French and English posts.
+		self::$model->post->save_translations( $posts['fr'], array( 'fr' => $posts['fr'], 'en' => $posts['en'] ) );
+
+		$terms = wp_get_object_terms( $posts, 'post_translations' );
+		$term_id = $terms[0]->term_id;
+		$this->assertCount( 2, $terms ); // Is correct at this step because the German post isn't translated into either English or French.
+
+		// Links again the German and English posts but not with the French one.
+		self::$model->post->save_translations( $posts['de'], array( 'de' => $posts['de'], 'en' => $posts['en'] ) );
+
+		$terms = wp_get_object_terms( $posts, 'post_translations' );
+		$this->assertCount( 1, $terms );
+		$this->assertSame( $term_id, $terms[0]->term_id ); // The term is the same as the latest save_translations().
```

Comment:
This assertion seems risky as you test one of 2 translation groups, and there is no warranty of the order. Moreover I don't see the added value: is it important that we kept the same term across all changes?